native/src/routes/news/components/News.js

```diff
@@ -41,63 +41,54 @@ export type PropsType = {|
 |}
 
 const News = (props: PropsType) => {
-  const {
-    news,
-    newsId,
-    cityCode,
-    language,
-    fetchMoreNews,
-    isFetchingMore,
-    selectedNewsType,
-    theme,
-    t,
-    routeKey
-  } = props
-  const { navigateTo, navigateToLink } = props
+  const { news, newsId, language, fetchMoreNews, isFetchingMore, selectedNewsType, theme, t, routeKey } = props
+  const { navigateTo, navigateToLink, cityModel } = props
 
-  const navigateToNews = useCallback(
-    (cityCode: string, language: string, newsId: string) => () => {
-      navigateTo({
-        route: NEWS_ROUTE,
-        cityCode,
-        languageCode: language,
-        newsId,
-        newsType: selectedNewsType
-      })
-    },
-    [selectedNewsType, navigateTo]
-  )
-
-  const renderNoItemsComponent = useCallback(() => {
+  const renderNoItemsComponent = () => {
     return <NoNews theme={theme}>{t('currentlyNoNews')}</NoNews>
-  }, [theme, t])
+  }
 
   const rendersNewsListItem = useCallback(
     (cityCode: string, language: string) => ({ item }: { item: LocalNewsModel | TunewsModel, ... }) => {
+      const navigateToNews = () => {
+        navigateTo({
+          route: NEWS_ROUTE,
+          cityCode,
+          languageCode: language,
+          newsId: item.id.toString(),
+          newsType: selectedNewsType
+        })
+      }
+
       return (
         <NewsListItem
           key={item.id}
           newsItem={item}
           language={language}
           theme={theme}
           isTunews={selectedNewsType === TU_NEWS_TYPE}
-          navigateToNews={navigateToNews(cityCode, language, item.id.toString())}
+          navigateToNews={navigateToNews}
           t={t}
         />
       )
     },
-    [selectedNewsType, navigateToNews, theme, t]
+    [selectedNewsType, theme, t, navigateTo]
   )
 
+  if (selectedNewsType === LOCAL_NEWS_TYPE ? !cityModel.pushNotificationsEnabled : !cityModel.tunewsEnabled) {
```

Comment:
for local news: why should ther be a NotFounderror if the pushNotifications aren't enabled?

Reply:
This is just shitty cms naming. This is whether the city has enabled local news or not. It has nothing (at least not directly) to do with actual push notifications.